src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,429 @@
+import { Gesture } from './gesture';
+import { GestureBuilder, BuiltGesture } from './gestureBuilder';
+import { Directions } from '../../Directions';
+import {
+  GestureEventPayload,
+  HandlerStateChangeEventPayload,
+} from '../gestureHandlerCommon';
+import { getNextHandlerTag } from '../handlersRegistry';
+import { PanGestureHandlerEventPayload } from '../PanGestureHandler';
+import { ForceTouchGestureHandlerEventPayload } from '../ForceTouchGestureHandler';
+import { PinchGestureHandlerEventPayload } from '../PinchGestureHandler';
+import { RotationGestureHandlerEventPayload } from '../RotationGestureHandler';
+
+type HitSlopType =
+  | number
+  | Partial<
+      Record<
+        'left' | 'right' | 'top' | 'bottom' | 'vertical' | 'horizontal',
+        number
+      >
+    >
+  | Record<'width' | 'left', number>
+  | Record<'width' | 'right', number>
+  | Record<'height' | 'top', number>
+  | Record<'height' | 'bottom', number>;
+
+type CommonGestureConfig = {
+  ref: React.RefObject<SimpleGesture>;
+  enabled: boolean;
+  minPointers: number;
+  shouldCancelWhenOutside: boolean;
+  hitSlop: HitSlopType;
+  requireToFail: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+  simultaneousWith: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+};
+
+type PanGestureHandlerEvent = GestureEventPayload &
+  PanGestureHandlerEventPayload;
+
+type PinchGestureHandlerEvent = GestureEventPayload &
+  PinchGestureHandlerEventPayload;
+
+type RotationGestureHandlerEvent = GestureEventPayload &
+  RotationGestureHandlerEventPayload;
+
+type ForceTouchGestureHandlerEvent = GestureEventPayload &
+  ForceTouchGestureHandlerEventPayload;
+
+type HandlerCallbacks = {
+  handlerTag: number;
+  onBegan?: (event: HandlerStateChangeEventPayload) => void;
+  onStart?: (event: HandlerStateChangeEventPayload) => void;
+  onEnd?: (event: HandlerStateChangeEventPayload, success: boolean) => void;
+  onUpdate?:
+    | ((event: PanGestureHandlerEvent) => void)
+    | ((event: PinchGestureHandlerEvent) => void)
+    | ((event: RotationGestureHandlerEvent) => void)
+    | ((event: ForceTouchGestureHandlerEvent) => void);
+};
+
+export abstract class SimpleGesture extends Gesture {
```

Comment:
I don't like this name. Why is it `Simple`?

Reply:
At the beginning, instead of the `GestureBuilder` there was a `ComplexGesture`, so `SimpleGesture` might have had more sense then. Would `BasicGesture` be better?

Reply:
`BaseGesture`?

Reply:
Changed in e6ab178da770372205b43669089ce73f8836288e